api/common_test.go

```diff
@@ -0,0 +1,175 @@
+package api
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+)
+
+type ports []types.Port
+
+// DisplayablePorts
+func TestDisplayablePorts(t *testing.T) {
+	cases := map[*ports]string{
```

Comment:
Small nit, but I found it a little weird to use a map this way. I have a slight preference for using a slice instead. In addition to making it clearer that this is being used as a list of test cases, it would make the test behave deterministically (map iteration is randomized).